### PR TITLE
Converts keycloakdeployment to a dependent resource

### LINKED ISSUE #22225: Convert KeycloakDeployment to a dependent resource
### Description

Due to some of the unit test logic and further refining how we're handling status and other logic that is now stateful in the KeycloakDeployment instance, this will be a more involved conversion.

https://github.com/operator-framework/java-operator-sdk/issues/1989 may be applicable here - not sure if this also ends up producing an extra revision like what was seen with secrets and ingress.

### PR DESCRIPTION
**Pull request by shawkins**
Other than the conversion to use a dependent resource (which moved the status handling to the controller which uses an annotation to track in progress migrations) the largest changes here were to break apart the responsibilities of KeycloakDistConfigurator.  The logic now just deals with the creation of env variables and tracking what are first class properties.  The rest of the logic was moved into / merged with the keycloak deployment dependent resource.  Hopefully this makes the env logic a little more straight-forward.

There are a couple of more "hacks" in the test logic to keep things basically the same it was but allow for CDI to provide the dependent resources.

Closes #22225

### CODE REVIEW

File: operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java:61]: This does not seem to be used anywhere.

File: operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java:87]: Keycloak tries to stay away from Mockito. Even though it might make sense to use it here, we'd be creating a precedence for the project. :) Could we perhaps try keeping the previous approach, i.e. implement a custom class implementing the Context interface?

**Review by vmuzikar — COMMENTED** [review]
@shawkins Sorry for the very late review. Added a few minor-ish comments. Other that that LGTM, nice work! I really needed this refactoring especially consolidating env vars logic.

File: operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
Comment (shawkins) [reply operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java:61]: It's used indirectly. With that annotation quarkus will use the mock bean in cdi.

File: operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
Comment (shawkins) [reply operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java:87]: > implement a custom class implementing the Context interface?

Isn't that just boiler plate for the sake of saying that mockito won't be used?  Wouldn't it be better to just enforce some principled use of Mockito?

File: operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
Comment (shawkins) [reply operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java:102]: >  OLM upgrade process installs all intermediate Operator versions when upgrading but AFAIK it doesn't assure reconciliation was finished at the intermediate installs.

I'm not sure either what assurances if any happen during the OLM upgrade.  I can remove the term upgrade here and rephrase as - once 21 falls out of any reasonable support, we don't need this.  At worst the user would need to install 22 first with manual approval to let reconcilations happen first.

File: operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
Comment (vmuzikar) [reply operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java:61]: Actually, it's needed by the `KeycloakDeploymentDependentResource`. Sorry for the false alarm. ;)

**Review by vmuzikar — COMMENTED** [review]
LGTM, if we're happy with keeping Mockito for an isolated test case.

File: operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java
Comment (vmuzikar) [reply operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java:102]: Technically, we support upgrade even from version 1 I think. :D

Comment (shawkins) [reply operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java]: Ok, just made the comment very general instead.

**Review by vmuzikar — APPROVED** [review]
@shawkins Thanks for the update!

**Pull request merged**
